microsoft/onefuzz
# Create autoscale resources for scaleset

## Summary of the Pull Request

_What is this about?_
This PR creates an auto scale resource for scalesets that uses the pool work queue as the tirgger metric.

## Info on Pull Request

_What does this include?_

* These changes only affect new scalesets, existing scalesets will not be modified
* Scalesets that already have an auto scale resource associated with them will not be modified
* The autoscaling resource is disabled meaning scaleset behavior won't change
* In the interest of keeping this reviewable, hooking up scaling event diagnostics will be a follow up PR

## Validation Steps Performed

_How does someone test & validate?_

1. Create a new scaleset
2. Find your new scaleset n azure portal
3. Under "Scaling", select the "JSON" tab
4. Verify there is 1 object under `.properties.profiles`
5. Verify `.properties.enabled == false`
